sdk/nodejs/x/automation/config.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export type ConfigValue = {
+    value: string;
+    secret?: boolean;
+};
```

Comment:
My philosophy here is that Automation API does not need to map every CLI concept. In this instance, structured config and `pulumi config set --path` are more important for human/cli interaction. Quite easy to read/write an entire key when you have a programming language in front of you rather than playing around with bash. 

We'll have the opportunity to address this later if we get strong feedback, but I don't want to worry about it for the time being.